From 2f8c261d19426430ccb3446f19ef9bf41a11cd77 Mon Sep 17 00:00:00 2001
From: Quanyang Wang <quanyang.wang@windriver.com>
Date: Mon, 6 Jan 2020 21:33:55 +0800
Subject: [PATCH] mtd: spi-nor: fix erasesize for dual qspi mode

The zcu102 and zc706 boards support qspi dual mode, and in
the dual mode, the erasesize for mtd device should be twice
as much as the erasesize for each nor flash chip.

Signed-off-by: Quanyang Wang <quanyang.wang@windriver.com>
Signed-off-by: Jiping Ma <jiping.ma2@windriver.com>
---
 drivers/mtd/spi-nor/spi-nor.c | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/drivers/mtd/spi-nor/spi-nor.c b/drivers/mtd/spi-nor/spi-nor.c
index 2e72bef5e1d2..22a53bc37e5f 100644
--- a/drivers/mtd/spi-nor/spi-nor.c
+++ b/drivers/mtd/spi-nor/spi-nor.c
@@ -4394,7 +4394,7 @@ static int spi_nor_select_erase(struct spi_nor *nor, u32 wanted_size)
 		if (!erase)
 			return -EINVAL;
 		nor->erase_opcode = erase->opcode;
-		mtd->erasesize = erase->size;
+		mtd->erasesize = erase->size << nor->shift;
 		return 0;
 	}
 
@@ -4412,7 +4412,7 @@ static int spi_nor_select_erase(struct spi_nor *nor, u32 wanted_size)
 	if (!erase)
 		return -EINVAL;
 
-	mtd->erasesize = erase->size;
+	mtd->erasesize = erase->size << nor->shift;
 	return 0;
 }
 
-- 
2.17.1

